### PR TITLE
Pycommon - reporter_config

### PR DESCRIPTION
**Pull request by cejkato2**
Many bugfixes and changes.
addressgroups still don't work (more specifically subnets don't work)

Netaddr dependency was removed, ipranges was used instead (because it is already required by idea-format).

File action flushes the file so it shouldn't be partially written.

### CODE REVIEW

File: pycommon/test/rc_22_addressgroup.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                Load mongo.yaml configuration file, parse it and analyze it
+
+                This shouldn't rise any exceptions
+
+                Should store message in DB and test if there is one record
+                """
+                self.config = Config(os.path.dirname(__file__) + '/rc_config/addressgroup.yaml');
+
+                self.assertNotEqual(self.config, None)
+                print("passing rules")
```
Comment (petrstehlik) [inline pycommon/test/rc_22_addressgroup.py]: Do these prints have any significant meaning or are just a leftover from debugging?

Comment (cejkato2) [inline pycommon/test/rc_22_addressgroup.py]: It is for debugging, this test class isn't finished yet. I have to solve an issue with IP subnets.

**Comment by cejkato2** [comment]
> Why was the unittest commented out? Does skipping a test cause something?

Yes, I observed issues with older versions of python.
Anyway, some tests are not necessary because we don't need to test external functionality. On the other hand, some useful basic tests of configuration, its parsing and evaluation were missing at all.

**Comment by cejkato2** [comment]
> Why is there a list of results? Is it used anywhere?

It is very useful and helpful for tests of rules. When I get list of "results" (if the condition of a rule is `true` or `false`), I can check how many rules were used and how.
It is necessary.e.g. for testing whether the `drop` action behaves according to specification.

**Comment by cejkato2** [comment]
last answer to @petrstehlik:
> IP ranges did work with netaddr correctly. Have you tried looking how the generated rule's string look like?

Generated rules (more precisely their conditions) look the same as from former version with netaddr.
I tried to update pynspect to 0.5 but without any change.

I think this feature is currently not supported by pynspect and it must be solved soon.

Luckily, we got rid of the netaddr dependency - hurray 👍

**Comment by cejkato2** [comment]
There are some issues related to `reporter_config` but they will be solved in some new PR.

**Pull request merged**